### ISSUE TITLE
Package "obexd-contentfilter-off" not displayed in "Installed applications"

### ISSUE THREAD

**Issue by Olf0** [mentaljam/harbour-storeman#140]
#### Observation
Although the RPM [obexd-contentfilter-off](https://openrepos.net/content/obexdcontentfilterolf/bluetooth-obex-filter) is installed (by Storeman, BTW), it not displayed in Storeman under "Installed applications".
If one searchs for it (e.g., per search term "Bluetooth") and selects it, it is correctly shown as installed.
#### Expected result
obexd-contentfilter-off is listed in "Installed applications".
Generalised: IMO all RPMs installed <s>from OpenRepos (or even better: not from Jolla)</s> should be listed in "Installed applications".
#### Importance
Since SailfishOS 3.4.0, when upgrading SailfishOS at the GUI, one is instructed to uninstall all RPMs, which replace "system packages" (i.e., one of the RPMs, which form the BaseOS of SailfishOS).  obexd-contentfilter-off is among them, but when a user looks for it in Storeman under "Installed applications" in order to uninstall it, it is not shown!  This likely also affects other installed RPMs from OpenRepos.
#### Analysis
Not much yet.  No idea, if this only affects certain OpenRepos-categories, or even which other packages may also not be displayed in "Installed applications", despite being installed per Storeman.

**Comment by Olf0** [comment]
### Not part of this issue
 
> > Generalised: IMO all RPMs installed from OpenRepos (or even better: not from Jolla) should be listed in "Installed applications".
> 
> I have already pointed in some of discussions here on GitHub, that I had not found a way to classify packages as installed **from** or **not from** OpenRepos. The best thing I can do is to filter installed applications that are available at enabled repositories.

Sorry, I should **not** have mentioned that: It is irrelevant for this bug report.  And I know that this is technically hard to impossible to address.  Thus I struck out the phrase "*from OpenRepos (or even better: not from Jolla)*" in the original report above.

> P.S. If I would filter installed applications that are not available at Jolla repositories then all system packages (that are both available at Jolla and OpenRepos) would not be shown as installed.

Oh, I never thought about that special case, if RPMs from OpenRepos replace system RPMs *by name*.
Side note 1: IMO that is a bad practice, naming a replacement for e.g. `openssh-1.2.3-abc.rpm` `openssh-joe-1.2.3-abc.rpm` (version numbers and release strings are not relevant for this example) should always be feasible.  The [combination of `Provides:`, `Obsoletes:` and `Conflicts:` statements to the original package](https://github.com/Olf0/obexd-contentfilter-off/blob/master/rpm/obexd-contentfilter-off.spec#L26) (an RPM replaces) is fully sufficient, the RPM name shall differ.
Side note 2: Using `libzypp` directly may provide the APIs needed to retrieve the information "which RPM comes from where", but you decided against switching the back-end long ago for reasons.  Hence I did not follow this route (i.e., researching it a bit) any further.

### Likely not an issue of Storeman

> Sorry, I cannot reproduce your issue. I have installed `obexd-contentfilter-off` and see it among the installed apps (see the screenshot in bottom of the comment).
> 
> So I guess that the `obexd-contentfilter-olf` repository is not enabled on your device and thus the `obexd-contentfilter-off` package is not shown as installed.

Correct. ;\
My bad, it was late at night and the first time I upgraded from SFOS 3.4.0 (here: to SFOS 4.0.1) at the GUI, I really should have carried out some more analyses before filing this issue.
The usability of the list Jolla's enhanced upgrade function in the SFOS settings, which displays all RPMs which replaced system RPMs on an SFOS installation, is low: It is unclear, if user input (e.g, tapping on list items) is expected, there is no feedback when something is done etc.
Supposedly it disabled or deleted the *obexd-contentfilter-olf* repository during the initial, preparatory, GUI based steps of the upgrade process (i.e., while one can still use Storeman in parallel), which easily explains my experience.

Unfortunately, I think I fixed this in the mean time (although I do not really remember to have done so): Now the *obexd-contentfilter-olf* repository is enabled and the `obexd-contentfilter-off` RPM displayed as expected in Storeman.

Sorry to have bothered you with this, looking more closely first of all this is an usability issue of Jolla's enhanced (per SFOS 3.4.0) GUI upgrader: It should make obvious, if user input is required (e.g. by highlighting selected items), and reveal when it does what (the point of time in these preparatory steps of the upgrade process, when the repositories offering RPMs which replace system RPMs are disabled or deleted, is unclear due to the lack of feedback)!
I will try to observe more closely what Jolla's GUI upgrader does exactly when, when using it the next time.

P.S.: Maybe this issue description and "resolution" can at least be helpful to be able to easily point others who observe this to an explanation.
